src/common/storage/src/stage.rs

```diff
@@ -247,6 +239,41 @@ impl StageFilesInfo {
         }
         Ok(files)
     }
+
+    /// Stat files concurrently.
+    pub async fn stat_concurrent(
+        &self,
+        operator: &Operator,
+        first_only: bool,
+        max_files: usize,
+        files: &Vec<String>,
+    ) -> Vec<Result<(String, Metadata)>> {
+        let mut futures = Vec::with_capacity(files.len());
+        let mut limit = 0;
+        for file in files {
```

Comment:
It's better to use a batch; we parallelize the batch. I'm wondering if there are too many files, will there be too many futures?
See: https://github.com/datafuselabs/databend/blob/3a50f2c918b527ed140c4f83e36a0aac01d5b31c/src/query/storages/fuse/src/io/files.rs#L63-L77

Reply:
Makes sense, let me change them into an iterator.